src/v/cluster/tests/autocreate_non_replicable_test.cc

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "cluster/metadata_cache.h"
+#include "cluster/shard_table.h"
+#include "cluster/simple_batch_builder.h"
+#include "cluster/tests/cluster_test_fixture.h"
+#include "cluster/topics_frontend.h"
+#include "cluster/types.h"
+#include "config/configuration.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "model/timeout_clock.h"
+#include "test_utils/async.h"
+#include "test_utils/fixture.h"
+
+#include <boost/test/tools/old/interface.hpp>
+
+#include <chrono>
+#include <vector>
+
+static const model::ns tp1 = model::ns("tp-1");
+static const model::ns tp2 = model::ns("tp-2");
+static const model::ns tp3 = model::ns("tp-3");
+
+std::vector<cluster::non_replicable_topic> test_topics() {
+    return std::vector<cluster::non_replicable_topic>{
+      cluster::non_replicable_topic{
+        .source = model::topic_namespace{test_ns, model::topic("test")},
+        .name = model::topic_namespace{tp1, model::topic("tp-1")}},
+      cluster::non_replicable_topic{
+        .source = model::topic_namespace{test_ns, model::topic("test")},
+        .name = model::topic_namespace{tp2, model::topic("tp-2")}},
+      cluster::non_replicable_topic{
+        .source = model::topic_namespace{test_ns, model::topic("test")},
+        .name = model::topic_namespace{tp3, model::topic("tp-3")}},
+    };
+}
+
+FIXTURE_TEST(
+  create_single_non_replicable_topic_test_at_current_broker,
+  cluster_test_fixture) {
+    model::node_id id{0};
+    auto app = create_node_application(id);
+    wait_for_controller_leadership(id).get();
+
+    std::vector<cluster::topic_result> results;
+    bool success = false;
+    while (!success) {
+        results = app->controller->get_topics_frontend()
+                    .local()
+                    .autocreate_non_replicable_topics(
+                      test_topics(), rpc::no_timeout)
+                    .get0();
+        success = std::all_of(
+          results.begin(), results.end(), [](const cluster::topic_result& r) {
+              return r.ec == cluster::errc::success;
+          });
+    }
+
+    BOOST_REQUIRE_EQUAL(results.size(), 3);
+}
+
+FIXTURE_TEST(
+  test_autocreate_non_replicable_on_non_leader, cluster_test_fixture) {
+    // root cluster node
+    model::node_id n_1(0);
+    model::node_id n_2(1);
+
+    // first controller
+    auto app_0 = create_node_application(n_1);
```

Comment:
How do you know `n_1` is not the leader? This test should always make a request on a node that is not the leader

Reply:
Something like this is probably required:
```
    auto leader
      = node_0->controller->get_partition_leaders().local().get_leader(
        model::controller_ntp);
```

Reply:
there's  `wait_for_controller_leadership(n_1).get();`